### PR TITLE
Lean: Translate function bodies that only consist of literals or variables

### PR DESCRIPTION
**Pull request by javra**
Translates
```sail
function foo(y : unit) -> unit = {
  return y
}
```
to
```lean
def foo (y : Unit) : Unit :=
  y

def initialize_registers : Unit :=
  ()
```

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
Nice. This looks sensible from the lean side.

File: test/lean/trivial.sail [hunk 1]
```diff
@@ -0,0 +1,3 @@
+function foo(y : unit) -> unit = {
+  return y
+}
```
Comment (tobiasgrosser) [inline test/lean/trivial.sail]: Add a newline at the end of this file. Github complains otherwise.

**Comment by javra** [comment]
I've added the trivial test file back to `test/lean/trivial.sail`, let's just collect Sail files there that can already be translated until we have automated testing.

**Review by tobiasgrosser — APPROVED** [review]

File: test/lean/trivial.sail (hunk 1 above)
Comment (tobiasgrosser) [reply test/lean/trivial.sail]: If sail supports comments, you could add the expected Lean output as a comment.

**Review by tobiasgrosser — APPROVED** [review]

**Comment by javra** [comment]
Could also add `trivial.expected.lean` with the expected content of the output, I think that's better for the future testing infrastructure and your IDE correctly identifies it as Lean code...

**Comment by tobiasgrosser** [comment]
> Could also add `trivial.expected.lean` with the expected content of the output, I think that's better for the future testing infrastructure and your IDE correctly identifies it as Lean code...

Indeed.

**Comment by javra** [comment]
Since the output is a whole Lean project it would be fitting for the expected output to be a whole project as well, but maybe that bloats the test directory too much?

**Comment by tobiasgrosser** [comment]
> Since the output is a whole Lean project it would be fitting for the expected output to be a whole project as well, but maybe that bloats the test directory too much?

I would leave this out for now.

File: src/sail_lean_backend/pretty_print_lean.ml [hunk 2]
```diff
@@ -65,7 +65,34 @@ let rec untuple_args_pat typs (P_aux (paux, ((l, _) as annot)) as pat) =
   | _, _ -> unreachable l __POS__ "Unexpected pattern/type combination"
 
 let doc_typ (Typ_aux (t, _) as typ) =
-  match t with Typ_id (Id_aux (Id "unit", _)) -> string "Unit" | _ -> failwith "Type not translatable yet."
+  match t with
+  | Typ_id (Id_aux (Id "unit", _)) -> string "Unit"
+  | Typ_id (Id_aux (Id "int", _)) -> string "Int"
```
Comment (bacam) [inline src/sail_lean_backend/pretty_print_lean.ml:70]: Note that there are several integer types, `int`, `atom`, `range`, and `implicit`.  The last two only appear in certain situations (actual ranges, and implicit arguments, respectively), but `atom` is the original name for "the integer that matches this type variable" and can appear unexpectedly.

**Review by bacam — APPROVED** [review]

**Comment by javra** [comment]
Can we merge this?

**Pull request merged**